core/src/test/scala/sttp/tapir/ShadowedEndpointCheckerTest.scala

```diff
@@ -0,0 +1,116 @@
+package sttp.tapir
+
+import org.scalatest.flatspec.AnyFlatSpecLike
+import org.scalatest.matchers.should.Matchers
+import sttp.tapir.util.ShadowedEndpointChecker
+
+class ShadowedEndpointCheckerTest extends AnyFlatSpecLike with Matchers {
+
+  it should "omit all segments which are not relevant for shadow check" in {
+    val e1 = endpoint.get.in(query[String]("key").and(header[String]("X-Account"))).in("x")
+    val e2 = endpoint.get.in("x")
+
+    val result = ShadowedEndpointChecker.apply(List(e1, e2))
```

Comment:
[minor] `apply` is special, you can write simply `ShadowedEndpointChecker(List(e1, e2))`